### PR TITLE
refactor(test): add scenario for test-stack-simple-nav

### PR DESCRIPTION
**Pull request by LKuchno**
## Description
Closes: https://github.com/software-mansion/react-native-screens-labs/issues/1601

This PR adds a comprehensive manual test scenario for the gamma/v5 `StackContainer`'s simple push/pop navigation, and adjusts the `test-stack-simple-nav` screen so the scenario can be exercised on both platforms.

The scenario documents the full lifecycle from root (**Home**) through stacking/unstacking screens **A** and **B**, covering:
- push/pop via the on-screen buttons,
- the native header back button,
- the iOS edge-swipe-back / Android system gesture-back.

It validates that each push creates a new instance with a unique `routeKey`, that a preserved (not recreated) screen keeps its `routeKey` on pop, and that the root screen (**Home**) cannot be popped. Edge cases (re-pushing an already-present route, rapid tapping push/pop, route-key uniqueness) are included.

The scenario also documents the platform launch difference: iOS runs from the in-app scenario menu, while Android must be launched **directly** via `apps/App.tsx` (rendering `TestStackSimpleNav` as the root). This is the current workaround for the Android back-button limitation ([issue #1459](https://github.com/software-mansion/react-native-screens-labs/issues/1459)), where the native back button and system gesture-back only pop the stack when the `StackContainer` is at the root rather than nested inside the example app's navigation.

## Changes

- **Test screen (`index.tsx`)**: Added `headerConfig` with a `title` to the **Home**, **A**, and **B** routes. Rendering a header makes the native back button appear on non-root screens (including on Android), which is required to test the header-back-button and gesture-back flows.
- **Scenario metadata (`scenario-description.ts`)**: Renamed the scenario from `"Simple navigation scenario"` to `"Simple stack navigation"`.
- **Scenario (`scenario.md`)**: Added the manual scenario documenting all push/pop flows, native header-back-button and edge-swipe / system gesture-back interactions, `routeKey` behavior, per-platform launch instructions, and edge cases (re-pushing the same route, rapid tapping, root-screen protection). E2E automation is noted as TBD.

### CODE REVIEW

File: apps/src/tests/single-feature-tests/stack-v5/test-stack-simple-nav/scenario.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+**Description:** Verify basic push and pop navigation on the gamma/v5
+`StackContainer`. The stack starts with **Home** and can push **A** or **B**
+any number of times (including re-pushing a route already on the stack),
+each push creating a new screen instance with its own unique `routeKey`;
+non-root screens expose a **Pop** button. The test validates that push/pop
+via the on-screen buttons, the iOS native header back button, and the iOS
+edge-swipe-back gesture all produce consistent stack state, that `routeKey`
+values are unique per pushed instance, and that the root screen (**Home**)
+cannot be popped. See the Notes for `routeKey` behavior and the Android
+back-button caveat (issue #1459).
```
[the remark this answers is not in the document]
Comment (LKuchno) [reply apps/src/tests/single-feature-tests/stack-v5/test-stack-simple-nav/scenario.md]: Updated with link to correct issue in rns-lab repo

File: apps/src/tests/single-feature-tests/stack-v5/test-stack-simple-nav/scenario.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+  the Android system/hardware back button does not currently pop the stack
+  (see issue #1459). All back-button steps below are therefore **iOS
+  only**; on Android use the on-screen **Pop** button to go back.
+
+## Steps
+
+### Baseline
+
+1. Launch the app and navigate to the **Simple navigation** screen (Stack
+   v5).
```
[the remark this answers is not in the document]
Comment (LKuchno) [reply apps/src/tests/single-feature-tests/stack-v5/test-stack-simple-nav/scenario.md]: https://github.com/software-mansion/react-native-screens/pull/4270/changes/b9461d508cf3325967dbd353e5ec48026393a6a1

File: apps/src/tests/single-feature-tests/stack-v5/test-stack-simple-nav/scenario.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+  the lifetime of the app session (it is shared across all Stack
+  containers). Do **not** expect specific numbers such as `r-A-1`: the
+  exact suffix depends on how many screens were pushed earlier in the
+  session and will differ between runs and after a JS reload. Only the
+  **relationships** matter — every push produces a strictly new `Key`, and
+  a preserved (not recreated) screen keeps the same `Key`.
+- **Android:** the gamma Stack does not render a header back button, and
+  the Android system/hardware back button does not currently pop the stack
+  (see issue #1459). All back-button steps below are therefore **iOS
+  only**; on Android use the on-screen **Pop** button to go back.
```
[the remark this answers is not in the document]
Comment (LKuchno) [reply apps/src/tests/single-feature-tests/stack-v5/test-stack-simple-nav/scenario.md]: https://github.com/software-mansion/react-native-screens/pull/4270/changes/ffa858781ec33c37be2ed156caeb04dcd515ebe7

**Review by kkafar — APPROVED** [review]
In general I'm against naming such details as background colours in the scenario description. It can very easily go out of date. 

This is a minor however. I recommend to think about it.
